File: zero-business/zero-main-business/zero-main-domain-business/src/main/scala/com/zjj/main/domain/user/UserRepository.scala
package com.zjj.main.domain.user

/**
 * @author zengJiaJun
 * @crateTime 2024年11月18日 21:06
 * @version 1.0
 */
trait UserRepository {

  def findById(id: Long): io.vavr.control.Option[UserAgg]
  def findByUserName(username: String): io.vavr.control.Option[UserAgg]
  def save(user: UserAgg): Unit
  def existsByRoles(roleIds: java.util.List[String]): Boolean
}

File: zero-business/zero-main-business/zero-main-domain-business/src/main/java/com/zjj/main/domain/user/UserRepository.java
package com.zjj.main.domain.user;

import io.vavr.control.Option;

import java.util.List;
import java.util.Set;

/**
 * @author zengJiaJun
 * @version 1.0
 * @crateTime 2025年03月24日 16:33
 */
public interface UserRepository {

    Option<UserAgg> findByUsername(String username);

    Option<UserAgg> findById(Long id);

    void save(UserAgg userAgg);


    List<UserAgg> queryByIds(Set<Long> ids);

    boolean existsByRoleId(Long roleId);
}
